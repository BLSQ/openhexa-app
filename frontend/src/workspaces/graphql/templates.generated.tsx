import * as Types from '../../graphql/types';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type GetPipelineVersionForTemplateQueryVariables = Types.Exact<{
  versionId: Types.Scalars['UUID']['input'];
}>;


export type GetPipelineVersionForTemplateQuery = { __typename?: 'Query', pipelineVersion?: { __typename?: 'PipelineVersion', id: string, zipfile: string } | null };


export const GetPipelineVersionForTemplateDocument = gql`
    query GetPipelineVersionForTemplate($versionId: UUID!) {
  pipelineVersion(id: $versionId) {
    id
    zipfile
  }
}
    `;

/**
 * __useGetPipelineVersionForTemplateQuery__
 *
 * To run a query within a React component, call `useGetPipelineVersionForTemplateQuery` and pass it any options that fit your needs.
 * When your component renders, `useGetPipelineVersionForTemplateQuery` returns an object from Apollo Client that contains loading, error, and data properties
 * you can use to render your UI.
 *
 * @param baseOptions options that will be passed into the query, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options;
 *
 * @example
 * const { data, loading, error } = useGetPipelineVersionForTemplateQuery({
 *   variables: {
 *      versionId: // value for 'versionId'
 *   },
 * });
 */
export function useGetPipelineVersionForTemplateQuery(baseOptions: Apollo.QueryHookOptions<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables> & ({ variables: GetPipelineVersionForTemplateQueryVariables; skip?: boolean; } | { skip: boolean; }) ) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useQuery<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>(GetPipelineVersionForTemplateDocument, options);
      }
export function useGetPipelineVersionForTemplateLazyQuery(baseOptions?: Apollo.LazyQueryHookOptions<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>) {
          const options = {...defaultOptions, ...baseOptions}
          return Apollo.useLazyQuery<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>(GetPipelineVersionForTemplateDocument, options);
        }
export function useGetPipelineVersionForTemplateSuspenseQuery(baseOptions?: Apollo.SkipToken | Apollo.SuspenseQueryHookOptions<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>) {
          const options = baseOptions === Apollo.skipToken ? baseOptions : {...defaultOptions, ...baseOptions}
          return Apollo.useSuspenseQuery<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>(GetPipelineVersionForTemplateDocument, options);
        }
export type GetPipelineVersionForTemplateQueryHookResult = ReturnType<typeof useGetPipelineVersionForTemplateQuery>;
export type GetPipelineVersionForTemplateLazyQueryHookResult = ReturnType<typeof useGetPipelineVersionForTemplateLazyQuery>;
export type GetPipelineVersionForTemplateSuspenseQueryHookResult = ReturnType<typeof useGetPipelineVersionForTemplateSuspenseQuery>;
export type GetPipelineVersionForTemplateQueryResult = Apollo.QueryResult<GetPipelineVersionForTemplateQuery, GetPipelineVersionForTemplateQueryVariables>;