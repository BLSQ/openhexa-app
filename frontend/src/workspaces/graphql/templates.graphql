query GetTemplateVersionForDownload($versionId: UUID!) {
  pipelineTemplateVersion(id: $versionId) {
    id
    versionNumber
    template {
      code
    }
    sourcePipelineVersion {
      zipfile
    }
  }
}
