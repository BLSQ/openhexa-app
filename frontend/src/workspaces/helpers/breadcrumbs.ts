/**
 * Utility functions for handling breadcrumb navigation in bucket file systems
 */

export interface BreadcrumbItem {
  label: string;
  value: string;
}

/**
 * Generates breadcrumb items from a prefix path
 * @param prefix - The current prefix path (e.g., "folder1/folder2/")
 * @returns Array of breadcrumb items with label and value for navigation
 */
export function generateBreadcrumbs(prefix: string | null): BreadcrumbItem[] {
  if (!prefix) return [];

  const breadcrumbs: BreadcrumbItem[] = [];
  let accumulated = "";

  prefix
    .split("/")
    .filter(Boolean)
    .forEach((part) => {
      accumulated = accumulated ? `${accumulated}/${part}` : part;
      breadcrumbs.push({
        label: part,
        value: `${accumulated}/`,
      });
    });

  return breadcrumbs;
}
