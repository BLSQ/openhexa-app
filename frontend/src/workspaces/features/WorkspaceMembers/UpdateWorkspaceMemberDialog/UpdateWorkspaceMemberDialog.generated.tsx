import * as Types from '../../../../graphql/types';

import { gql } from '@apollo/client';
export type UpdateWorkspaceMember_WorkspaceMemberFragment = { __typename?: 'WorkspaceMembership', id: string, role: Types.WorkspaceMembershipRole };

export const UpdateWorkspaceMember_WorkspaceMemberFragmentDoc = gql`
    fragment UpdateWorkspaceMember_workspaceMember on WorkspaceMembership {
  id
  role
}
    `;