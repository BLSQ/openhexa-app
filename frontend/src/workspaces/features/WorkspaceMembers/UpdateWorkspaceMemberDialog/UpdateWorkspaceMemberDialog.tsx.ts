import { graphql } from "graphql/gql";

export const UpdateWorkspaceMemberWorkspaceMemberDoc = graphql(`
fragment UpdateWorkspaceMember_workspaceMember on WorkspaceMembership {
  id
  role
}
`);
