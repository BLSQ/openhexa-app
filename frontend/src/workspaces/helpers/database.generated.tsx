import * as Types from '../../graphql/types';

import { gql } from '@apollo/client';
import * as Apollo from '@apollo/client';
const defaultOptions = {} as const;
export type DeleteWorkspaceDatabaseTableMutationVariables = Types.Exact<{
  input: Types.DeleteWorkspaceDatabaseTableInput;
}>;


export type DeleteWorkspaceDatabaseTableMutation = { __typename?: 'Mutation', deleteWorkspaceDatabaseTable?: { __typename?: 'DeleteWorkspaceDatabaseTableResult', success: boolean, errors: Array<Types.DeleteWorkspaceDatabaseTableError> } | null };


export const DeleteWorkspaceDatabaseTableDocument = gql`
    mutation deleteWorkspaceDatabaseTable($input: DeleteWorkspaceDatabaseTableInput!) {
  deleteWorkspaceDatabaseTable(input: $input) {
    success
    errors
  }
}
    `;
export type DeleteWorkspaceDatabaseTableMutationFn = Apollo.MutationFunction<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>;

/**
 * __useDeleteWorkspaceDatabaseTableMutation__
 *
 * To run a mutation, you first call `useDeleteWorkspaceDatabaseTableMutation` within a React component and pass it any options that fit your needs.
 * When your component renders, `useDeleteWorkspaceDatabaseTableMutation` returns a tuple that includes:
 * - A mutate function that you can call at any time to execute the mutation
 * - An object with fields that represent the current status of the mutation's execution
 *
 * @param baseOptions options that will be passed into the mutation, supported options are listed on: https://www.apollographql.com/docs/react/api/react-hooks/#options-2;
 *
 * @example
 * const [deleteWorkspaceDatabaseTableMutation, { data, loading, error }] = useDeleteWorkspaceDatabaseTableMutation({
 *   variables: {
 *      input: // value for 'input'
 *   },
 * });
 */
export function useDeleteWorkspaceDatabaseTableMutation(baseOptions?: Apollo.MutationHookOptions<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>) {
        const options = {...defaultOptions, ...baseOptions}
        return Apollo.useMutation<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>(DeleteWorkspaceDatabaseTableDocument, options);
      }
export type DeleteWorkspaceDatabaseTableMutationHookResult = ReturnType<typeof useDeleteWorkspaceDatabaseTableMutation>;
export type DeleteWorkspaceDatabaseTableMutationResult = Apollo.MutationResult<DeleteWorkspaceDatabaseTableMutation>;
export type DeleteWorkspaceDatabaseTableMutationOptions = Apollo.BaseMutationOptions<DeleteWorkspaceDatabaseTableMutation, DeleteWorkspaceDatabaseTableMutationVariables>;