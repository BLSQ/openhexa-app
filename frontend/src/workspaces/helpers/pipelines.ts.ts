import { graphql } from "graphql/gql";

export const NewRunDoc = graphql(`
fragment NewRun on PipelineRun {
  id
}
`);
