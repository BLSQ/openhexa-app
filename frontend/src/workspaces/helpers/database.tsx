import { gql } from "@apollo/client";
import { getApolloClient } from "core/helpers/apollo";
import {
  DeleteWorkspaceDatabaseTableMutation,
  DeleteWorkspaceDatabaseTableMutationVariables,
} from "./database.generated";

type lang = "R" | "PYTHON";

export const getReadTableSnippet = (tableName: string) => {
  return `import os
import pandas as pd
from sqlalchemy import create_engine

engine = create_engine(os.environ["WORKSPACE_DATABASE_URL"])

pd.read_sql("SELECT * FROM ${tableName}", con=engine)`;
};

export const getUsageSnippet = (tableName: string, lang: lang) => {
  let text = "";
  switch (lang) {
    case "PYTHON":
      text = `import os
import pandas as pd
from sqlalchemy import create_engine

engine = create_engine(os.environ["WORKSPACE_DATABASE_URL"])

# Create sample dataframe
df = pd.DataFrame({"name": ["Jane", "John", "Tyler"], "age": [19, 17, 22]})

# Write data
df.to_sql("${tableName}", con=engine, if_exists="replace")`;
      break;
    case "R":
      text = `library(DBI)
  
con <- dbConnect(
    RPostgres::Postgres(),
    dbname = Sys.getenv("WORKSPACE_DATABASE_DB_NAME"),
    host = Sys.getenv("WORKSPACE_DATABASE_HOST"),
    port = Sys.getenv("WORKSPACE_DATABASE_PORT"),
    user = Sys.getenv("WORKSPACE_DATABASE_USERNAME"),
    password = Sys.getenv("WORKSPACE_DATABASE_PASSWORD")
)

dbWriteTable(con, "${tableName}", Data_fin, overwrite=TRUE)`;
      break;
  }
  return text;
};

export async function deleteTable(workspace: string, tableName: string) {
  const client = getApolloClient();
  const { data } = await client.mutate<
    DeleteWorkspaceDatabaseTableMutation,
    DeleteWorkspaceDatabaseTableMutationVariables
  >({
    mutation: gql`
      mutation deleteWorkspaceDatabaseTable(
        $input: DeleteWorkspaceDatabaseTableInput!
      ) {
        deleteWorkspaceDatabaseTable(input: $input) {
          success
          errors
        }
      }
    `,
    variables: { input: { workspaceSlug: workspace, table: tableName } },
  });

  if (!data?.deleteWorkspaceDatabaseTable?.success) {
    throw new Error("Impossible to delete table");
  }
  return true;
}
