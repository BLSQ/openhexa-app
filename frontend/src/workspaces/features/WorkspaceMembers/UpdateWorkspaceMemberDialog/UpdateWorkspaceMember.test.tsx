import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import {
  UpdateWorkspaceMemberDocument,
  useUpdateWorkspaceMemberMutation,
} from "workspaces/graphql/mutations.generated";
import { TestApp } from "core/helpers/testutils";
import router from "next/router";
import UpdateWorkspaceMemberDialog from ".";
import { v4 } from "uuid";
import { WorkspaceMembershipRole } from "graphql/types";

const MEMBER = {
  id: v4(),
  role: WorkspaceMembershipRole.Viewer,
};

jest.mock("workspaces/graphql/mutations.generated", () => ({
  ...jest.requireActual("workspaces/graphql/mutations.generated"),
  __esModule: true,
  useUpdateWorkspaceMemberMutation: jest.fn().mockReturnValue([]),
}));

const useUpdateWorkspaceMemberMutationMock =
  useUpdateWorkspaceMemberMutation as jest.Mock;

describe("UpdateWorkspaceMemberDialog", () => {
  const onClose = jest.fn();
  beforeEach(() => {
    useUpdateWorkspaceMemberMutationMock.mockClear();
  });

  it("is not displayed ", async () => {
    const { container } = render(
      <TestApp mocks={[]}>
        <UpdateWorkspaceMemberDialog
          open={false}
          member={MEMBER}
          onClose={() => {}}
        />
      </TestApp>,
    );
    const dialog = await screen.queryByRole("dialog");
    expect(dialog).not.toBeInTheDocument();
    expect(onClose).not.toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  it("is displayed when open is true", async () => {
    const { container } = render(
      <TestApp mocks={[]}>
        <UpdateWorkspaceMemberDialog
          open={true}
          member={MEMBER}
          onClose={() => {}}
        />
      </TestApp>,
    );

    const dialog = await screen.queryByRole("dialog");
    expect(dialog).toBeInTheDocument();
    expect(onClose).not.toHaveBeenCalled();
    expect(container).toMatchSnapshot();
  });

  it("Update a workspace member ", async () => {
    const pushSpy = jest.spyOn(router, "push");
    const { useUpdateWorkspaceMemberMutation } = jest.requireActual(
      "workspaces/graphql/mutations.generated",
    );
    useUpdateWorkspaceMemberMutationMock.mockImplementation(
      useUpdateWorkspaceMemberMutation,
    );

    const user = userEvent.setup();
    const mocks = [
      {
        request: {
          query: UpdateWorkspaceMemberDocument,
          variables: {
            input: {
              membershipId: MEMBER.id,
            },
          },
        },
        result: {
          data: {
            updateWorkspaceMember: {
              success: true,
              errors: [],
            },
          },
        },
      },
    ];
    render(
      <TestApp mocks={mocks}>
        <UpdateWorkspaceMemberDialog
          open={true}
          member={MEMBER}
          onClose={() => {}}
        />
      </TestApp>,
    );
    expect(useUpdateWorkspaceMemberMutationMock).toHaveBeenCalled();

    const updateButton = screen.getByRole("button", { name: "Save" });
    await user.click(updateButton);
  });
});
