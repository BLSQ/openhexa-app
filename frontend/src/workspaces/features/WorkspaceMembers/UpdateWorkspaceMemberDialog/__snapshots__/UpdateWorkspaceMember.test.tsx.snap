// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UpdateWorkspaceMemberDialog is displayed when open is true 1`] = `<div />`;

exports[`UpdateWorkspaceMemberDialog is not displayed  1`] = `<div />`;
